### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by HollyFredD** [AdguardTeam/AdGuardHome#3032]
### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details
AdGuard Home
Version: v0.106.0
Channel: release
Go version: go1.15.11
Build time: 2021-04-28T14:09:50Z+0000
GOOS: linux
GOARCH: amd64
Race: false
Main module:
        github.com/AdguardTeam/AdGuardHome (devel)
Dependencies:
        github.com/AdguardTeam/dnsproxy@v0.37.2 (sum: h1:3lgizD+lZI6uqxFiQykd1/hV7Ji4vSJBMejl1rbFAXU=)
        github.com/AdguardTeam/golibs@v0.4.5 (sum: h1:RRA9ZsmbJEN4OllAx0BcfvSbRBxxpWluJijBYmtp13U=)
        github.com/AdguardTeam/urlfilter@v0.14.5 (sum: h1:WyF0hg0MwKevsqNPkoaZFH8f5WRi/yuy/7qePtYt5Ts=)
        github.com/NYTimes/gziphandler@v1.1.1 (sum: h1:ZUDjpQae29j0ryrS0u/B8HZfJBtBQHjqw2rQ2cqUQ3I=)
        github.com/aead/chacha20@v0.0.0-20180709150244-8b13a72661da (sum: h1:KjTM2ks9d14ZYCvmHS9iAKVt9AyzRSqNU1qabPih5BY=)
        github.com/aead/poly1305@v0.0.0-20180717145839-3fee0db0b635 (sum: h1:52m0LGchQBBVqJRyYYufQuIbVqRawmubW3OFGqK1ekw=)
        github.com/ameshkov/dnscrypt/v2@v2.1.3 (sum: h1:DG4Uf7LSDg6XDj9sp3maxh3Ur26jeGQaP5MeYosn6v0=)
        github.com/ameshkov/dnsstamps@v1.0.3 (sum: h1:Srzik+J9mivH1alRACTbys2xOxs0lRH9qnTA7Y1OYVo=)
        github.com/beefsack/go-rate@v0.0.0-20200827232406-6cde80facd47 (sum: h1:M57m0xQqZIhx7CEJgeLSvRFKEK1RjzRuIXiA3HfYU7g=)
        github.com/cheekybits/genny@v1.0.0 (sum: h1:uGGa4nei+j20rOSeDeP5Of12XVm7TGUd4dJA9RDitfE=)
        github.com/digineo/go-ipset/v2@v2.2.1 (sum: h1:k6skY+0fMqeUjjeWO/m5OuWPSZUAn7AucHMnQ1MX77g=)
        github.com/fsnotify/fsnotify@v1.4.9 (sum: h1:hsms1Qyu0jgnwNXIxa+/V/PDsU6CfLf6CNO8H7IWoS4=)
        github.com/go-ping/ping@v0.0.0-20210216210419-25d1413fb7bb (sum: h1:2opwLSXqxE0Za64PdpskXuvLYDj/XHQAD8tLcYpSlvY=)
        github.com/gobuffalo/envy@v1.9.0 (sum: h1:eZR0DuEgVLfeIb1zIKt3bT4YovIMf9O9LXQeCZLXpqE=)
        github.com/gobuffalo/packd@v1.0.0 (sum: h1:6ERZvJHfe24rfFmA9OaoKBdC7+c9sydrytMg8SdFGBM=)
        github.com/gobuffalo/packr@v1.30.1 (sum: h1:hu1fuVR3fXEZR7rXNW3h8rqSML8EVAf6KNm0NKO/wKg=)
        github.com/google/renameio@v1.0.1-0.20210406141108-81588dbe0453 (sum: h1:vvZyWjAX9oDB+DWpMsZMyv6Q3NZtim2C5Zcdh+H0OmQ=)
        github.com/AdguardTeam/dhcp@v0.0.0-20210420175708-50b0efd52063 (sum: h1:RBsQppxEJEqHApY6WDBkM2H0UG5wt57RcT0El2WGdp8=)
        github.com/joho/godotenv@v1.3.0 (sum: h1:Zjp+RcGpHhGlrMbJzXTrZZPrWj+1vfm90La1wgB6Bhc=)
        github.com/joomcode/errorx@v1.0.3 (sum: h1:3e1mi0u7/HTPNdg6d6DYyKGBhA5l9XpsfuVE29NxnWw=)
        github.com/josharian/native@v0.0.0-20200817173448-b6b71def0850 (sum: h1:uhL5Gw7BINiiPAo24A2sxkcDI0Jt/sqp1v5xQCniEFA=)
        github.com/kardianos/service@v1.2.0 (sum: h1:bGuZ/epo3vrt8IPC7mnKQolqFeYJb7Cs8Rk4PSOBB/g=)
        github.com/lucas-clemente/quic-go@v0.20.1 (sum: h1:hb5m76V8QS/8Nw/suHvXqo3BMHAozvIkcnzpJdpanSk=)
        github.com/marten-seemann/qtls-go1-15@v0.1.4 (sum: h1:RehYMOyRW8hPVEja1KBVsFVNSm35Jj9Mvs5yNoZZ28A=)
        github.com/mdlayher/ethernet@v0.0.0-20190606142754-0394541c37b7 (sum: h1:lez6TS6aAau+8wXUP3G9I3TGlmPFEq2CTxBaRqY6AGE=)
        github.com/mdlayher/netlink@v1.4.0 (sum: h1:n3ARR+Fm0dDv37dj5wSWZXDKcy+U0zwcXS3zKMnSiT0=)
        github.com/mdlayher/raw@v0.0.0-20191009151244-50f2db8cc065 (sum: h1:aFkJ6lx4FPip+S+Uw4aTegFMct9shDvP+79PsSxpm3w=)
        github.com/miekg/dns@v1.1.40 (sum: h1:pyyPFfGMnciYUk/mXpKkVmeMQjfXqt3FAJ2hy7tPiLA=)
        github.com/patrickmn/go-cache@v2.1.0+incompatible (sum: h1:HRMgzkcYKYpi3C8ajMPV8OFXaaRUnok+kx1WdO15EQc=)
        github.com/pkg/errors@v0.9.1 (sum: h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=)
        github.com/rogpeppe/go-internal@v1.7.0 (sum: h1:3qqXGV8nn7GJT65debw77Dzrx9sfWYgP0DDo7xcMFRk=)
        github.com/satori/go.uuid@v1.2.0 (sum: h1:0uYX9dsZ2yD7q2RtLRtPSdGDWzjeM3TbMJP9utgA0ww=)
        github.com/ti-mo/netfilter@v0.4.0 (sum: h1:rTN1nBYULDmMfDeBHZpKuNKX/bWEXQUhe02a/10orzg=)
        github.com/u-root/u-root@v7.0.0+incompatible (sum: h1:u+KSS04pSxJGI5E7WE4Bs9+Zd75QjFv+REkjy/aoAc8=)
        go.etcd.io/bbolt@v1.3.5 (sum: h1:XAzx9gjCb0Rxj7EoqcClPD1d5ZBxZJk0jbuoPHenBt0=)
        golang.org/x/crypto@v0.0.0-20210220033148-5ea612d1eb83 (sum: h1:/ZScEX8SfEmUGRHs0gxpqteO5nfNW6axyZbBdw9A12g=)
        golang.org/x/net@v0.0.0-20210405180319-a5a99cb37ef4 (sum: h1:4nGaVu0QrbjT/AK2PRLuQfQuh6DJve+pELhqTdAj3x0=)
        golang.org/x/sys@v0.0.0-20210330210617-4fbd30eecc44 (sum: h1:Bli41pIlzTzf3KEY06n+xnzK/BESIg2ze4Pgfh/aI8c=)
        golang.org/x/text@v0.3.5 (sum: h1:i6eZZ+zk0SOf0xgBpEpPD18qWcJda6q1sxt3S0kzyUQ=)
        gopkg.in/natefinch/lumberjack.v2@v2.0.0 (sum: h1:1Lc07Kr7qY4U2YPouBjpCLxpiyxIVoxqXgkXLknAOE8=)
        gopkg.in/yaml.v2@v2.4.0 (sum: h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=)
        howett.net/plist@v0.0.0-20201203080718-1454fab16a06 (sum: h1:QDxUo/w2COstK1wIBYpzQlHX/NqaQTcf9jyz347nI58=)

### Expected Behavior
Before upgrading to v0.106.0, DNS Rewrite was working properly to resolve internal domains.

### Actual Behavior
After the upgrade, local domains are not resolved anymore.

**Comment by ainar-g** [comment]
Hi, and thank you for the report.  What exactly do you mean by “local domains are not resolved anymore”?  Could you add an example of a query that doesn't work any more?  For example, using `dig` or `nslookup`?  Thanks.

**Comment by HollyFredD** [comment]
Hi,

By internal domains I mean as an example "local.lan" for which correspond to "192.168.0.x"

**Comment by ainar-g** [comment]
The local domain name (`lan` by default) is applied to the DHCP leases by default.  Do you run AGH as your network's DHCP server?  If so, could you please check if your AGH is affected by bug #3027?

**Comment by HollyFredD** [comment]
I am using my router as a DHCP server, not AGH.

**Comment by ainar-g** [comment]
Does changing `dns.local_domain_name` from `lan` to something else fix the issue?  We have a theory regarding this behaviour.

**Comment by HollyFredD** [comment]
Unfortunately, I cannot try it out now. I reverted back my AGH instance to 105.2 as it is used in production. (in my house, but yet complicated to debug while being yeld at by all family members because nothing is working properly :-) )

**Comment by boot-ini** [comment]
I'm also having this issue and yes, changing `dns.local_domain_name` to `local` fixed it :)

**Comment by posixx** [comment]
Hi, where do you change dns.local_domain_name? i cannot find this setting. I'm using adguard home snap image and DHCP disabled on adguard..

**Comment by bmartron** [comment]
Hello,
i also have the same issue, so where can i change dns.local_domain_name ?
In my router (also dhcp).
Thanks

**Comment by boot-ini** [comment]
It's in your `AdGuardHome.yaml`:

```dns:
  local_domain_name: local```

There may be consequences to changing it, though

**Comment by ainar-g** [comment]
If you're not running AGH as the DHCP server, than there really shouldn't be any consequences.

Also, this issue is probably caused by the same bugs as #3028, and we've recently pushed some fixes for that one.  Perhaps this issue could be marked as duplicate?

**Comment by bmartron** [comment]
Thanks for your reply.
i also can say "this is working" if i change local_domain_name from lan to local.
hope this is help to solved the issue when user is using another dhcp server.
Thanks again

**Comment by malcinator** [comment]
> It's in your `AdGuardHome.yaml`:
> 
> ```
> dns:
>   local_domain_name: local
> ```
> 
> There may be consequences to changing it, though

There are. It will then not resolve .local address which as a macOS user is also a problem

**Comment by ainar-g** [comment]
@HollyFredD We released v0.106.1 today.  Your issue should be solved.  Can you place check it?

**Comment by HollyFredD** [comment]
> 
> 
> @HollyFredD We released v0.106.1 today. Your issue should be solved. Can you place check it?

Hi, I've just updated to v0.106.1 and looks like it works again. (at least for the few pings I tried after a flushDNS).
Thanks for your reactivity and prompt solution on that one !

From my end, I consider this ticket can be considered as closed/resolved.

Thanks !
